OlxProject/pages/home_page.rb

```diff
@@ -0,0 +1,32 @@
+require_relative '../tools/all_requires'
+require_relative 'top_part'
+
+class HomePage
+  include TopPart
+
+  protected def home_search_field
+    driver.find_element(id: 'headerSearch')
+  end
+
+  protected def home_search_button
+    driver.find_element(id: 'submit-searchmain')
+  end
+
+  def home_search_field_input(search_item)
+    home_search_field.send_keys search_item
+  end
+
+  def home_search_button_click
+    home_search_button.click
+  end
+
+  def make_success_home_search
+    home_search_button_click
+    SuccessfulSearchPage.new
+  end
+
+  def make_unsuccess_home_search
+    home_search_button_click
+    UnsuccessfulSearchPage.new
+  end
```

Comment:
these two methods look weird. Both just perform clicks but with different results. As a user of this code I will be expecting to do HomePage.new.make_unsuccess_home_search to get to UnsuccessfulSearchPage? It seems this method doesn't really control the behavior of success or unsuccessful of search, but the data/steps in test do.

Reply:
ok, I`ll update this. I`ll make 1 method, which make search and return success/unsuccess page depending on result.